packages/valory/skills/abstract_round_abci/behaviour_utils.py

```diff
@@ -358,7 +358,10 @@ def __handle_waiting_for_message(self) -> None:
                 self.__notified = False
                 self.__message = None
         else:
-            self.__get_generator_act().send(None)
+            try:
+                self.__get_generator_act().send(self.__message)
+            except StopIteration:
+                self.__handle_stop_iteration()
```

Comment:
This is more readable:
https://github.com/valory-xyz/open-autonomy/pull/1996#discussion_r1236967696